yjit/src/core.rs

```diff
@@ -377,6 +378,158 @@ fn get_iseq_payload(iseq: IseqPtr) -> &'static mut IseqPayload
     unsafe { payload_non_null.as_mut() }.unwrap()
 }
 
+/// Free the per-iseq YJTI payload
+#[no_mangle]
+pub extern "C" fn rb_yjit_iseq_free(payload: *mut c_void)
+{
+    let payload = {
+        if payload.is_null() {
+            // Nothing to free.
+            return;
+        } else {
+            payload as *mut IseqPayload
+        }
+    };
```

Comment:
Detail maybe, but you could pass an IseqPtr to these functions instead. Then you'd have access to the `get_iseq_payload` method which does the unwrapping, and you'd be able to print the iseq pointer for debugging (if ever needed).

Reply:
`get_iseq_payload` allocates the payload when it's null though so to make that change I would need to touch some other functions -- wouldn't want to be allocating and then immediately freeing the payload. I think it's a good call to make the change for other reasons but it feels out of scope for this PR.